network/records/subnets.go

```diff
@@ -105,7 +105,14 @@ func SharedSubnets(a, b []byte, maxLen int) []int {
 	if len(a) == 0 || len(b) == 0 {
 		return shared
 	}
-	for subnet, aval := range a {
+
+	// Fix https://github.com/ssvlabs/ssv/issues/1802
+	limit := len(a)
+	if len(b) < limit {
+		limit = len(b)
+	}
+
+	for subnet, aval := range a[:limit] {
```

Comment:
i think we should require that both arrays are of len == 128 (i think we have a constant for 128)

otherwise, what does an array of 64 bytes mean? are the rest of the 64 subnets 0 or 1? i think if its not specified then its an invalid subnets array

Reply:
Yes, that's what I think too, let's refactor this in another PR. I've reverted the initial fix so I'm not going to touch this function in this PR at all

Reply:
@moshe-blox refactored in https://github.com/ssvlabs/ssv/pull/1813, please review it